standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreChecker.java

```diff
@@ -425,11 +425,11 @@ void findUnknownPartitions(Table table, Set<Path> partPaths,  Set<Path> partPath
     Set<Path> correctPartPathsInMS = new HashSet<>(partPathsInMS);
     // remove partition paths in partPathsInMS, to getPartitionsNotOnFs
     partPathsInMS.removeAll(allPartDirs);
-    FileSystem fs = tablePath.getFileSystem(conf);
     // There can be edge case where user can define partition directory outside of table directory
     // to avoid eviction of such partitions
     // we check for partition path not exists and add to result for getPartitionsNotOnFs.
     for (Path partPath : partPathsInMS) {
+      FileSystem fs = partPath.getFileSystem(conf);
```

Comment:
This could be costly to do it every time.
Are we expecting this to be different for every `partPath`?